core/src/value/person_name.rs

```diff
@@ -0,0 +1,326 @@
+    EmptyGiven {},
+}
+pub type Result<T, E = Error> = std::result::Result<T, E>;
+
+/// Represents a Dicom PersonName.
+/// Contains family, given, middle name, prefix and suffix.
+/// For practical purposes, family and given name are mandatory components, not optional as per standard.
+/// The Dicom string representation is split by "^" separator into its respective components.
+/// Does not support component group delimiter "=", ideographic and phonetic characters.
+#[derive(Debug, Clone, PartialEq)]
+pub struct PersonName<'a> {
+    prefix: Option<&'a str>,
+    family: &'a str,
+    middle: Option<&'a str>,
+    given: &'a str,
+    suffix: Option<&'a str>,
+}
+
+impl<'a> From<PersonName<'a>> for PrimitiveValue {
+    fn from(p: PersonName) -> Self {
+        dicom_value!(Str, p.to_dicom_string())
+    }
+}
+
+macro_rules! write_if_some {
+    ($formater: expr, $option: expr,  $format: expr) => {
+        if let Some(opt) = $option {
+            write!($formater, $format, opt)?
+        }
+    };
+}
+
+impl Display for PersonName<'_> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write_if_some!(f, self.prefix, "{} ");
+        write!(f, "{} ", self.family)?;
+        write_if_some!(f, self.middle, "{} ");
+        write!(f, "{}", self.given)?;
+        write_if_some!(f, self.suffix, " {}");
+        Ok(())
+    }
+}
+
+impl<'a> PersonName<'a> {
+    pub fn new(
+        prefix: Option<&'a str>,
+        given: &'a str,
+        family: &'a str,
+        middle: Option<&'a str>,
+        suffix: Option<&'a str>,
+    ) -> Self {
+        Self {
+            prefix,
+            family,
+            middle,
+            given,
+            suffix,
+        }
+    }
+    pub fn to_dicom_string(&self) -> String {
+        let mut name = format!("{}^{}", self.family, self.given);
+        match self.middle {
+            Some(middle) => name.push_str(&format!("^{}", middle)),
+            None => {
+                if self.prefix.is_some() || self.suffix.is_some() {
+                    name.push('^')
+                }
+            }
+        }
+        match self.prefix {
+            Some(prefix) => match self.suffix {
+                Some(suffix) => name.push_str(&format!("^{}^{}", prefix, suffix)),
+                None => name.push_str(&format!("^{}", prefix)),
+            },
+            None => match self.suffix {
+                Some(suffix) => name.push_str(&format!("^^{}", suffix)),
+                None => {}
+            },
+        }
+        name
+    }
+    pub fn prefix(&self) -> Option<&str> {
+        self.prefix
+    }
+    pub fn suffix(&self) -> Option<&str> {
+        self.suffix
+    }
+    pub fn family(&self) -> &str {
+        self.family
+    }
+    pub fn given(&self) -> &str {
+        self.given
+    }
+    pub fn middle(&self) -> Option<&str> {
+        self.middle
+    }
+
+    // exact match to dicom formatted string
+    pub fn is(&self, search: &str) -> bool {
```

Comment:
This method is a bit confusing and expensive. Maybe one should rather invite the user into converting both operands to a common type: either both `PersonName` or both strings. Then the semantics are much clearer. Otherwise, they may be lured into calling this method multiple times assuming it to be a plain component-by-component check when it actually converts `self` into a string underneath.